### PR TITLE
Tweaks for deployment

### PR DESCRIPTION
**Pull request by percyliang**
Trying again...

Made a bunch of changes while trying to deploy to worksheets-dev.codalab.org, including passing in more config options, fixing a bug with SSL/nginx config, cleanup the bundle store, bug with command-line parsing, and general cleanup.

### CODE REVIEW

File: docs/Server-Setup.md [hunk 1]
```diff
@@ -133,29 +133,17 @@ You can also start an instance and run tests on it:
...
-    virtualenv -p python3.6 venv3.6
-    venv3.6/bin/pip install black
-    venv3.6/bin/black codalab worker scripts *.py --diff
-
 These must pass before you submit a PR.
 
-## Other
-
-To auto-generate the REST reference and CLI references:
+## Pre-commit
```
Comment (epicfaace) [inline docs/Server-Setup.md:138]: We can (later) integrate this with https://pre-commit.com/

File: codalab/rest/workers.py [hunk 2]
```diff
@@ -27,11 +27,11 @@ def checkin(worker_id):
     socket_id = local.worker_model.worker_checkin(
         request.user.user_id,
         worker_id,
-        request.json["tag"],
-        request.json["cpus"],
-        request.json["gpus"],
-        request.json["memory_bytes"],
-        request.json["free_disk_bytes"],
+        request.json.get("tag"),
```
Comment (epicfaace) [inline codalab/rest/workers.py:32]: Why do we need this? Does request.json sometimes not have the right keys?

File: docker/compose_files/files/nginx.conf [hunk 3]
```diff
@@ -99,19 +98,6 @@ http {
     #proxy_read_timeout 1200;
     #send_timeout 1200;
 
-    location /bundleservice {
```
Comment (epicfaace) [inline docker/compose_files/files/nginx.conf:102]: Why did you decide to merge /bundleservice with the root url / ? What did we have at / before?

File: pre-commit.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  venv2.7/bin/pip install -r requirements-server.txt || exit 1
+  venv2.7/bin/pip install -r requirements.docs.txt || exit 1
+  rm -rf worker/codalabworker.egg-info  # Need to clear because of different Python versions
+  venv2.7/bin/pip install -e worker  # Not sure why this is necessary
+  venv2.7/bin/pip install -e .
+fi
+
+if ! [ -e venv3.6 ]; then
+  virtualenv -p python3.6 venv3.6 || exit 1
+  venv3.6/bin/pip install black || exit 1
```
Comment (epicfaace) [inline pre-commit.sh]:
```suggestion
  venv3.6/bin/pip install black==18.9b0 || exit 1
```

File: pre-commit.sh [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+# TODO: merge into one once we have Python 3.
+
+if ! [ -e venv2.7 ]; then
+  virtualenv -p python2.7 venv2.7 || exit 1
+  venv2.7/bin/pip install -r requirements-server.txt || exit 1
+  venv2.7/bin/pip install -r requirements.docs.txt || exit 1
+  rm -rf worker/codalabworker.egg-info  # Need to clear because of different Python versions
+  venv2.7/bin/pip install -e worker  # Not sure why this is necessary
+  venv2.7/bin/pip install -e .
```
Comment (epicfaace) [inline pre-commit.sh:14]: Why are we installing codalab etc. before each commit? (it seems like the main purpose of ./pre-commit.sh might be just to fix style and generate docs?)

**Review by epicfaace — APPROVED** [review]

File: docker/compose_files/files/nginx.conf (hunk 3 above)
Comment (percyliang) [reply docker/compose_files/files/nginx.conf:102]: `/bundleservice` no longer exists - this is from the distant past.

File: codalab/rest/workers.py (hunk 2 above)
Comment (percyliang) [reply codalab/rest/workers.py:32]: Old workers might not have all the fields, but we don't want to crash.  Added a comment.

File: pre-commit.sh (hunk 5 above)
Comment (percyliang) [reply pre-commit.sh:14]: Generating docs requires having a running version of CodaLab (since it needs to load all the libraries).

**Pull request merged**